### PR TITLE
Make `shell_executable=None` pass raw commands but default to `sh -c` wrap

### PR DESCRIPTION
**Pull request by KuxaBeast**
Allow setting `shell_executable` in ssh connector to `None` to pass raw commands instead of wrapping it as a payload to the `shell_executable`. Default to `sh` as default shell for backward compatibility. Solves [#692](https://github.com/Fizzadar/pyinfra/issues/692).

### CODE REVIEW

**Comment by KuxaBeast** [comment]
Okay, I implemented the base but it seems like it defaults to `shell_executable=None` instead of `sh`, which I set in the function arguments as default, and I can't find where it is actually overridden.

**Review by Fizzadar — APPROVED** [review]
Thank you for pulling this together @KuxaBeast!

**Comment by KuxaBeast** [comment]
Thanks, @Fizzadar. I think it still requires some polishing though. There is still the issue of the default behaviour. As I previously stated, I don't have a clue where is it overridden, because the default value in the function definition didn't seem to do anything.

File: pyinfra/api/connectors/util.py [hunk 1]
```diff
@@ -328,12 +325,12 @@ def make_unix_command(
 
         if shell_executable is not None:
             # Quote the whole shell -c 'command' as BSD `su` does not have a shell option
-            command_bits.append(QuoteString(StringCommand(shell_executable, '-c', command)))
+            command_bits.append(QuoteString(StringCommand(shell_executable, '-c', QuoteString(command))))
         else:
             command_bits.append(StringCommand(command))
```
Comment (KuxaBeast) [inline pyinfra/api/connectors/util.py]: I might have forgotten `QuoteString()` around the `StringCommand()` here.

**Comment by Fizzadar** [comment]
@KuxaBeast my apologies - I jumped the gun a bit here! So the default value is populated here:

https://github.com/Fizzadar/pyinfra/blob/current/pyinfra/api/operation_kwargs.py#L69

Which itself is retrieved from the config, with the default `None` here:

https://github.com/Fizzadar/pyinfra/blob/current/pyinfra/api/config.py#L55

I think we should be safe to change the `Config.SHELL` default and have this behave as expected.

---

For full context any of the "global arguments", such as `shell_executable`, can be passed directly into operation/deploy calls, defined in the config and also as host data. The config value is the default and in most cases this is never used.

**Comment by KuxaBeast** [comment]
Okay, I think I found another problem. When I changed the default of `Config.SHELL` variable to `sh` another problem appeared. Until now, I have been verifying the behaviour using the `exec` operation and the `--shell-executable` argument, but it seems it accepts any value as string, resulting in `None -c command` or `null -c command`.

Calling: `$ pyinfra -vv --debug yum.ospf exec --shell-executable=None -- '/file print'`
results in:
```
[yum.ospf] >>> None -c '/file print'
[yum.ospf] expected command name (line 1 column 1)
```

I am not sure why didn't I notice it before, but the `server.shell` operation does not return the output of executed commands for me, but does indeed support setting the `None` value by passing `shell_executable=null` as the second argument. (`--shell-executable=null` doesn't work with the `exec` op.)

Successful, but does not print command output:
`$ pyinfra -vv --debug yum.ospf server.shell '/file print' shell_executable=null`

File: pyinfra/api/connectors/util.py (hunk 1 above)
Comment (KuxaBeast) [reply pyinfra/api/connectors/util.py]: Yup. Totally forgot. Will fix in next commit.

**Comment by Fizzadar** [comment]
Using `-vvv` with the second version will print the output.

Exec is a bit of a special case, under the hood it actually calls `server.shell` and sets the print output flags. I think it's acceptable to parse `'None'` into `None` when handling the CLI arg (https://github.com/Fizzadar/pyinfra/blob/current/pyinfra_cli/main.py#L461).

This works with `server.shell` because the arguments are parsed as JSON such that `'null'` becomes `None`, etc.

**Comment by KuxaBeast** [comment]
Oh, I see now. It might be a good idea to mention the JSON-parsing in the docs somewhere. Anyway, I am gonna explicitly convert `'None'` and `'null'` arguments to `None` in the pyinfra_cli/main.py for now. Maybe this would deserve a whole separate cli flag like `--no-shell` instead?

File: pyinfra/api/connectors/util.py [hunk 2]
```diff
@@ -240,7 +240,7 @@ def make_unix_command(
     command,
     env=None,
     chdir=None,
-    shell_executable='sh',
+    shell_executable=None,
```
Comment (Fizzadar) [inline pyinfra/api/connectors/util.py]: I think this should remain `'sh'` for now as the default and `None`/no shell has to be explicitly set.

**Review by Fizzadar — COMMENTED** [review]
One suggestion - everytihng else looks good!

On the CLI flag I think it's best to keep one arg for the purpose, converting from the string values seems acceptable to me (I think, open to thoughts!). Agreed this needs to be better documented!

File: pyinfra/api/connectors/util.py (hunk 2 above)
Comment (KuxaBeast) [reply pyinfra/api/connectors/util.py]: Okay. Fixing.

**Review by Fizzadar — APPROVED** [review]
Thank you once again for pulling this together @KuxaBeast, really exciting to start opening up pyinfra to different target types!

EDIT: I pushed a tiny fix for some tests checking for the call to `make_unix_command` so I can get this into `v1.6`!

**Comment by vojta001** [comment]
I wish I had known this earlier. The docs aren't really detailed in this regard.

Anyway, is there a way to set this option in the inventory? `{"_shell_executable": None}` is of no luck for me.

**Comment by Fizzadar** [comment]
> I wish I had known this earlier. The docs are really detailed in this regard.
> 
> Anyway, is there a way to set this option in the inventory? `{"_shell_executable": None}` is of no luck for me.

Hrm that should work inside the inventory, mind filing a bug report for that?

**Comment by vojta001** [comment]
@Fizzadar Sure! Thank you for your dedication

**Pull request merged**